packages/smooth_app/lib/generic_lib/dialogs/smooth_alert_dialog.dart

```diff
@@ -84,10 +85,19 @@ class SmoothAlertDialog extends StatelessWidget {
     );
   }
 
-  Padding _buildBottomBar(EdgeInsets padding) {
+  Padding _buildBottomBar(EdgeInsetsGeometry padding) {
+    final double topPadding;
+    if (padding is EdgeInsetsDirectional) {
+      topPadding = padding.bottom;
+    } else if (padding is EdgeInsets) {
+      topPadding = padding.bottom;
+    } else {
+      topPadding = 0.0;
+    }
```

Comment:
Cannot the point of this: there's only one place this method is called, and with a fat chance we need `padding.bottom` anyway.
Maybe I'm missing something here.

Reply:
I would like to force `EdgeInsetsDirectional` everywhere, but here indeed we have still strange behavior.

Before reverting this part, what do you think about forcing all `EdgeInsets` to be `EdgeInsetsDirectional` and enforcing this with a Lint rule?

Reply:
For the moment just make `padding` an `EdgeInsetsDirectional`, which will give you access to `.bottom` and enable you to remove your tests.

> Before reverting this part, what do you think about forcing all EdgeInsets to be EdgeInsetsDirectional and enforcing this with a Lint rule?

If it's easy to you to code and easy for us to maintain, why not.
But it's not a priority.

Reply:
I don't know why I've overengineered this part.
That should be better now.

For the Lint part, yes indeed, not a priority